src/device-state/target-state.ts

```diff
@@ -76,16 +76,24 @@ const emitTargetState = (
 		!cachedResponse.emitted &&
 		emitter.listenerCount('target-state-update') > 0
 	) {
+		// CachedResponse has not been emitted before so emit as an update
 		emitter.emit(
 			'target-state-update',
 			_.cloneDeep(cachedResponse.body),
 			force,
 			isFromApi,
 		);
-
+		return true;
```

Comment:
Don't think I'm being overcritical or overthinking stuff. Couldn't we get the same objective by emitting a `fresh` flag with the `target-state-update` event? Something like

```
const emitTargetState = (
	cachedResponse: CachedResponse,
	force = false,
	isFromApi = false,
): boolean => {
	if (
		!cachedResponse.emitted &&
		emitter.listenerCount('target-state-update') > 0
	) {
		// CachedResponse has not been emitted before so emit as an update
		emitter.emit(
			'target-state-update',
			_.cloneDeep(cachedResponse.body),
			force,
			isFromApi,
                        true // is fresh
		);
		return true;
	} 
        emitter.emit(
	     'target-state-update',
	     _.cloneDeep(cachedResponse.body),
	     force,
	     isFromApi,
             !!cache.emitted // fresh will be false if we already emitted
        );
       
	// Return the same emitted value but normalized to be a boolean
	return !!cache.emitted;
};
```

And then in the handler do

```
if (fresh) {
    // Only save fresh target
    await deviceState.setTarget(targetState);
}
deviceState.triggerApplyTarget({ force, isFromApi });
```

If just feel that the target-state module having to care whether the request comes from the API or not and know not to emit the target state is getting very complex. What do you think?

Reply:
I mean this is fine, and probably we should not keep spending time on this. We can make a note to revisit this module to clean it up eventually

Reply:
I'm totally fine with discussing better ways to write the fix :p 

This is nice because we don't have to add a new emit type but then it sounds redundant. The name of this event implies that the data is fresh since it contains the word "update". I think a good solution is changing the event name to `target-state-notify` which then a listener can apply some business logic and emit `target-state-update` to run deviceState.setTarget and `target-state-apply` to run deviceState.triggerApplyTarget. 

It sounds like we just made it more complicated but it's more flexible now and easier to write new functionality that listens on specific events. Plus the event names are 100% exactly what's happening and not a combination of actions.

Reply:
I think of it from the point of view of the single responsibilty principle. The target state module has the responsibility to query the target state endpoint, and report the results. It should not care where the request came from, it probably shouldn't even care if there is someone listening. That last one something I added to fix #1455, but now feels like a hack as well. Having the module now be aware of whether an target state should be applied or just notified feels outside scope.

I think this partly has to do with the supervisor design, where you start a bunch of services performing actions on a timer and sending messages and have sometimes to perform extra checks to ensure that events don't catch you in an inconsistent state. That is probably something we could look to improve in the medium term.